File: code-examples/python/coupons/getbyid.py
import http.client

conn = http.client.HTTPConnection("api.marketcloud.it")

headers = {
    'authorization': "YourPublicKey:YourToken",
    'accept': "application/json",
    'content-type': "application/json"
    }

conn.request("GET", "/v0/coupons/1", headers=headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))

File: code-examples/java/coupons/getbyid.java
HttpResponse<String> response = Unirest.get("http://api.marketcloud.it/v0/coupons/1")
  .header("authorization", "YourPublicKey:YourToken")
  .header("accept", "application/json")
  .header("content-type", "application/json")
  .asString();